### PR TITLE
Add discover provider feature

### PR DESCRIPTION
**Pull request by GabrielSVinha**
Create discovery class and responds method that returns the response if a given IP and port represents a LXCA appliance

### CODE REVIEW

**Comment by rodneyhbrown7** [comment]
This looks good Gabriel. Can you add a spec test for this feature?

**Comment by GabrielSVinha** [comment]
Yep. Working on it.

File: lib/xclarity_client/discover.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        response = conf.get do |req|
+          req.url RESOURCE
+          req.options.timeout = 5           # open/read timeout in seconds
+          req.options.open_timeout = 6      # connection open timeout in seconds
+        end
+      rescue Exception
+        return false
+      end
+
+      return false if %w(200 302).include? response.status
```
Comment (AndreyMenezes) [inline lib/xclarity_client/discover.rb]: Here!

Comment (AndreyMenezes) [inline lib/xclarity_client/discover.rb]: True and false here was switched.

File: spec/xclarity_client_discover_spec.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+require 'spec_helper'
+
+describe XClarityClient::Discover do
+  context '#responds?' do
+    context 'without the correct appliance IPAddress' do
+      it 'should return false' do
+        res = XClarityClient::Discover.responds?(Faker::Internet.ip_v4_address, Faker::Number.number(4))
```
Comment (AndreyMenezes) [inline spec/xclarity_client_discover_spec.rb]: It's better to have a readable name in this variable. "res"?

File: spec/xclarity_client_discover_spec.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        expect(res).not_to be_truthy
+      end
+    end
+
+    context 'with the correct appliance IPAddress' do
+      before do
+        @port = Faker::Number.number(4)
+        @address = URI('https://' + Faker::Internet.ip_v4_address + ':' + @port)
+        WebMock.allow_net_connect!
+        stub_request(:get, File.join(@address.to_s, '/aicc')).to_return(:status => [200, 'OK'])
```
Comment (AndreyMenezes) [inline spec/xclarity_client_discover_spec.rb]: I think we need test 200 and 302 to have a more complete test.

**Comment by AndreyMenezes** [comment]
LGTM
Good comments!

**Pull request merged**